### PR TITLE
[hotfix] Remove local timeout for Hybrid and Tiered result partition test.

### PR DESCRIPTION
**Pull request by reswqa**
## What is the purpose of the change

*`HsResultPartitionTest#testRelease` and `TieredResultPartitionTest#testRelease`  depend on local timeout, which will make the test unstable.*

## Brief change log

  - *Remove local timeout for Hybrid and Tiered result partition test.*

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/util/concurrent/IgnoreShutdownRejectedExecutionHandler.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    public IgnoreShutdownRejectedExecutionHandler(Logger logger) {
+        this.logger = logger;
+    }
+
+    @Override
+    public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
+        if (executor.isShutdown()) {
+            if (logger != null) {
+                logger.debug("Execution rejected because shutdown is in progress");
```
Comment (TanYuxin-tyx) [inline flink-core/src/main/java/org/apache/flink/util/concurrent/IgnoreShutdownRejectedExecutionHandler.java:47]: "Execution is rejected" may be more appropriate.

File: flink-core/src/main/java/org/apache/flink/util/concurrent/IgnoreShutdownRejectedExecutionHandler.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+
+/**
+ * Rejected executions are ignored or logged in debug if the executor is {@link
+ * java.util.concurrent.ThreadPoolExecutor#isShutdown shutdown}. Otherwise, {@link
+ * RejectedExecutionException} is thrown.
+ */
+public class IgnoreShutdownRejectedExecutionHandler implements RejectedExecutionHandler {
+    private final Logger logger;
```
Comment (TanYuxin-tyx) [inline flink-core/src/main/java/org/apache/flink/util/concurrent/IgnoreShutdownRejectedExecutionHandler.java:33]: We'd better mark this logger as Nullable

**Review by TanYuxin-tyx — APPROVED** [review]
@reswqa Thanks for the fix. 
LGTM, only left some minor comments, PTAL.

File: flink-core/src/main/java/org/apache/flink/util/concurrent/IgnoreShutdownRejectedExecutionHandler.java (hunk 2 above)
Comment (reswqa) [reply flink-core/src/main/java/org/apache/flink/util/concurrent/IgnoreShutdownRejectedExecutionHandler.java]: Marked.

**Comment by reswqa** [comment]
Thanks @TanYuxin-tyx for the quick review, I have updated and force pushed this as it is only minor fix.

**Pull request merged**